src/Compilers/CSharp/Portable/Binder/Binder_Operators.cs

```diff
@@ -3205,6 +3213,11 @@ private BoundExpression BindAsOperator(BinaryExpressionSyntax node, DiagnosticBa
                 operand = new BoundDefaultExpression(defaultLiteral.Syntax, constantValueOpt: ConstantValue.Null,
                     type: GetSpecialType(SpecialType.System_Object, diagnostics, node));
             }
+            else if (operand.IsTypelessNew())
+            {
+                operand = CreateImplicitNewConversion(operand.Syntax, operand, Conversion.ImplicitNew,
+                    isCast: false, destination: resultType, diagnostics);
+            }
 
```

Comment:
Why did we switch the behavior for target-typed new?  I think of it as more akin to a lambda expression (`(x => 1) as Func<int>` is an error) than a null literal.  Did the LDM suggest a change here?

Reply:
If you wanted to give it a type the new expression has a place for the type to go.

---
In reply to: [241912281](https://github.com/dotnet/roslyn/pull/29167#discussion_r241912281) [](ancestors = 241912281)

Reply:
Changed in this PR, but it's been too long I don't remember where it's been suggested.
default and null both allowed, but I'd agree it's closer to lambda conversion. will revert.